### PR TITLE
Add a GitHub workflow to build and publish Docker images

### PR DESCRIPTION
**Pull request by mnonnenmacher**
See the commit messages for details.

The workflow was tested in a fork: https://github.com/mnonnenmacher/ort-server
The resulting Docker images can be seen here (they are mixed with test artifacts, only those starting with "ort-server-" are relevant): https://github.com/mnonnenmacher?tab=packages

### CODE REVIEW

**Comment by sschuberth** [comment]
FYI, I just realized your commits are not signed-off. Apparently they don't have to be... but should we still set up the DCO bot?

**Comment by mnonnenmacher** [comment]
> FYI, I just realized your commits are not signed-off. Apparently they don't have to be... but should we still set up the DCO bot?

Interesting, I assume it's not required because the `eclipsefdn/eca` verifies that the contributor agreement was signed, but I have to read up on it.

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+  build:
+    name: Build Docker Image
+    runs-on: ubuntu-22.04
+    permissions:
+      packages: write
+    strategy:
```
Comment (nnobelis) [inline .github/workflows/docker-build.yml:19]: Some idea:
Use max-parallel to build some images in parallel ? 
Use fail-fast: false to build all images even if some of them fail ?

File: .github/actions/free-disk-space/action.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Print Disk Space
+      shell: bash
+      run: df -h
+    - name: List Docker Images
+      if: ${{ false }} # Can be enabled if the 'Remove Unneeded Docker Images' step below needs to be updated.
```
Comment (nnobelis) [inline .github/actions/free-disk-space/action.yml:13]: What does this switch do ? Is is to disable the step ?

File: .github/actions/free-disk-space/action.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      shell: bash
+      run: docker images
+    - name: Remove Unneeded Docker Images
+      shell: bash
+      run: |
```
Comment (nnobelis) [inline .github/actions/free-disk-space/action.yml:18]: Is clearing some of this an unoptimization ? If for instance ubuntu:22.04 is required for build as  base image, it now has to be downloaded everytime ?

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (mnonnenmacher) [reply .github/workflows/docker-build.yml:19]: > Use max-parallel to build some images in parallel ?

That's already the case, they are all built in parallel.

> Use fail-fast: false to build all images even if some of them fail ?

I think if one build fails there is not much use for the other images.  I would rather go in the opposite direction and prevent the publishing of all images if a single one cannot be built, but not in this PR.

File: .github/actions/free-disk-space/action.yml (hunk 3 above)
Comment (mnonnenmacher) [reply .github/actions/free-disk-space/action.yml:18]: If we would use any of those images, yes, but I don't think we ever will.

**Review by bs-ondem — APPROVED** [review]

**Pull request merged**